voltdb/datadog_checks/voltdb/check.py

```diff
@@ -0,0 +1,148 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, List, Optional, cast
+
+import pkg_resources
+import requests
+from six import raise_from
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.db import Query, QueryManager
+
+from . import queries
+from .config import Config
+from .types import Instance
+
+BASE_PARSED_VERSION = pkg_resources.get_distribution('datadog-checks-base').parsed_version
+
+
+class VoltDBCheck(AgentCheck):
+    __NAMESPACE__ = 'voltdb'
+
+    def __init__(self, name, init_config, instances):
+        # type: (str, dict, list) -> None
+        super(VoltDBCheck, self).__init__(name, init_config, instances)
+        self._config = Config(cast(Instance, self.instance))
+
+        if self._config.auth is not None:
+            password = self._config.auth._password
+            self.register_secret(password)
+
+        manager_queries = [
+            queries.CPUMetrics,
+            queries.MemoryMetrics,
+            queries.SnapshotStatusMetrics,
+            queries.CommandLogMetrics,
+            queries.ProcedureMetrics,
+            queries.LatencyMetrics,
+            queries.StatementMetrics,
+            queries.GCMetrics,
+            queries.IOStatsMetrics,
+            queries.TableMetrics,
+            queries.IndexMetrics,
+        ]
+
+        if BASE_PARSED_VERSION < pkg_resources.parse_version('15.0.0'):
+            # On Agent < 7.24.0 we must to pass `Query` objects instead of dicts.
+            manager_queries = [Query(query) for query in manager_queries]  # type: ignore
```

Comment:
Nice, this is useful! But in the future if this check gets pinned to >15, won't this line be not-applicable since older Agent versions would be incompatible?

Reply:
Yes, probably, though I guess we'll be able to see this compat code and remove it eventually if/when bumping to 15+. For now we need the integration to be 7.22+ compatible since that's what the customer is using.